nalgebra-euler-angles/src/euler_angles.rs

```diff
@@ -0,0 +1,183 @@
+    let normalized_angle = degrees_angle % 360.0;
+
+    let radians_angle = normalized_angle * std::f64::consts::PI / 180.0;
+
+    let sin = radians_angle.sin();
+    let cos = radians_angle.cos();
+
+    return vec![
+        vec![cos, 0.0, sin],
+        vec![0.0, 1.0, 0.0],
+        vec![-sin, 0.0, cos]
+    ];
+}
+
+
+
+pub fn rotation_z(degrees_angle: &f64) -> Vec<Vec<f64>> {
+    let normalized_angle = degrees_angle % 360.0;
+
+    let radians_angle = normalized_angle * std::f64::consts::PI / 180.0;
+
+    let sin = radians_angle.sin();
+    let cos = radians_angle.cos();
+
+    return vec![
+        vec![cos, -sin, 0.0],
+        vec![sin, cos, 0.0],
+        vec![0.0, 0.0, 1.0]
+    ];
+}
+
+
+pub fn euler_to_rotation_matrix(degrees_angles: &[f64; 3], order: &[char; 3]) -> Result<Vec<Vec<f64>>, String> {
+    for chr in order {
+        if !matches!(*chr, 'X' | 'Y' | 'Z') {
+            let err_msg = format!("Invalid order character '{}'. Expected X/Y/Z", chr);
+            return Err(err_msg);
+        }
+    }
+
+    let rx = rotation_x(&degrees_angles[0]);
+    let ry = rotation_y(&degrees_angles[1]);
+    let rz = rotation_z(&degrees_angles[2]);
+
+
+
+    let mut result = vec![
+        // Identity matrix
+        vec![1.0, 0.0, 0.0],
+        vec![0.0, 1.0, 0.0],
+        vec![0.0, 0.0, 1.0]
+    ];
+
+
+
+
+    for &axis in order {
+        match axis {
+            'X' => result = multiply(&result, &rx),
+            'Y' => result = multiply(&result, &ry),
+            'Z' => result = multiply(&result, &rz),
+            _ => return Err(format!("Unexpected axis '{}'", axis)),
+        }
+    }
+
+    Ok(result)
+}
+
+
+
+use std::collections::HashMap;
+
+
+pub fn rotation_matrix_to_euler(rotation_matrix: &Vec<Vec<f64>>, order: &[char; 3]) -> Result<Vec<f64>, String> {
+    // Validate the order
+    for chr in order {
+        if !matches!(*chr, 'X' | 'Y' | 'Z') {
+            let err_msg = format!("Invalid order character '{}'. Expected X/Y/Z", chr);
+            return Err(err_msg);
+        }
+    }
+
+    let beta = -rotation_matrix[2][0].asin();                                                     // Y-axis rotation (beta)
+    let alpha = (rotation_matrix[2][1] / beta.cos()).atan2(rotation_matrix[2][2] / beta.cos());   // X-axis rotation (alpha)
+    let gamma = (rotation_matrix[1][0] / beta.cos()).atan2(rotation_matrix[0][0] / beta.cos());   // Z-axis rotation (gamma)
+
+    // Create a map to associate axes with their corresponding Euler angle functions
+    let mut euler_map: HashMap<char, f64> = HashMap::new();
+
+    euler_map.insert('X', alpha);
+    euler_map.insert('Y', beta);
+    euler_map.insert('Z', gamma);
+
+    let mut euler_angles = Vec::new();
+
+    for &axis in order {
+        match euler_map.get(&axis) {
+            Some(&angle) => euler_angles.push(angle), 
+            None => return Err(format!("Unexpected axis '{}'", axis)),
+        }
+    }
+
+    Ok(euler_angles)
+}
```

Comment:
Is the using the extrinsic or extrinsic definition? And how to discern between those?